wix-embedded-mysql/src/test/scala/com/wix/mysql/EmbeddedMysql8Test.scala

```diff
@@ -0,0 +1,281 @@
+        .addSchema("aSchema",
+          aMigrationWith("create table t1 (col1 INTEGER);"),
+          aMigrationWith("create table t2 (col1 INTEGER);"))
+      )
+
+      aQuery(mysqld, onSchema = "aSchema", sql = "select count(col1) from t1;") must beSuccessful
+      aQuery(mysqld, onSchema = "aSchema", sql = "select count(col1) from t2;") must beSuccessful
+    }
+
+    "apply migrations via SchemaConfig" in {
+      val config = aSchemaConfig("aSchema")
+        .withScripts(
+          aMigrationWith("create table t1 (col1 INTEGER);"),
+          aMigrationWith("create table t2 (col1 INTEGER);"))
+        .withCommands(
+          "create table t3 (col1 INTEGER);\n" +
+            "create table t4 (col2 INTEGER)")
+        .build
+
+      val mysqld = start(anEmbeddedMysql(testConfigBuilder(v8_latest).build).addSchema(config))
+
+      aQuery(mysqld, onSchema = "aSchema", sql = "select count(col1) from t1;") must beSuccessful
+      aQuery(mysqld, onSchema = "aSchema", sql = "select count(col1) from t2;") must beSuccessful
+      aQuery(mysqld, onSchema = "aSchema", sql = "select count(col1) from t3;") must beSuccessful
+      aQuery(mysqld, onSchema = "aSchema", sql = "select count(col2) from t4;") must beSuccessful
+    }
+
+    "quote created table name" in {
+      val config = aSchemaConfig("a-table")
+        .withScripts(aMigrationWith("create table t1 (col1 INTEGER);"))
+        .build
+
+      val mysqld = start(anEmbeddedMysql(testConfigBuilder(v8_latest).build).addSchema(config))
+
+      aQuery(mysqld, onSchema = "a-table", sql = "select count(col1) from t1;") must beSuccessful
+    }
+
+  }
+
+  "EmbeddedMysql schema modification" should {
+
+    "drop existing schema" in {
+      val schemaConfig = aSchemaConfig("aSchema").build
+
+      val mysqld = start(anEmbeddedMysql(testConfigBuilder(v8_latest).build).addSchema(schemaConfig))
+
+//      mysqld must haveSchemaCharsetOf(UTF8MB4, schemaConfig.getName)
+
+      mysqld.dropSchema(schemaConfig)
+
+      mysqld must notHaveSchema(schemaConfig.getName)
+    }
+
+    "fail on dropping of non existing schema" in {
+      val schemaConfig = aSchemaConfig("aSchema").build
+
+      val mysqld = start(anEmbeddedMysql(testConfigBuilder(v8_latest).build))
+
+      mysqld must notHaveSchema(schemaConfig.getName)
+
+      mysqld.dropSchema(schemaConfig) must throwA[CommandFailedException]
+    }
+/*
+    "add schema after mysqld start" in {
+      val schemaConfig = aSchemaConfig("aSchema").build
+
+      val mysqld = start(anEmbeddedMysql(testConfigBuilder(v8_latest).build))
+
+      mysqld must notHaveSchema(schemaConfig.getName)
+
+      mysqld.addSchema(schemaConfig)
+
+      mysqld must haveSchemaCharsetOf(UTF8MB4, schemaConfig.getName)
+    }
+*/
+    "fail on adding existing schema" in {
+      val schemaConfig = aSchemaConfig("aSchema").build
+
+      val mysqld = start(anEmbeddedMysql(testConfigBuilder(v8_latest).build).addSchema(schemaConfig))
```

Comment:
It will be safer to also add OS env var
`LD_LIBRARY_PATH=${embedmysql}/bin:${embedmysql}/lib:${embedmysql}/lib/private:.`
to make sure mysqld can find and load the **so** files.

8.0.19's **mysqld** does not search for current directory for any deps. This behavior is different from 8.0.17 at least. I assume 8.0.20 will be the same as 8.0.19.

Reply:
I would put that in a subsequent patch